### DISCUSSION TITLE
where to find more about http2_keep_alive_while_idle

### DISCUSSION THREAD

**Discussion by AlexXi19** [hyperium/tonic#1402] [Q&A]
I'm suspecting a connection issues with an upstream crate that uses tonic and traced the keep alive variable to the `http2_keep_alive_while_idle` field. However as I was looking through the source code, I wasn't able to find more information like how this is used. `// Set http2 KEEP_ALIVE_INTERVAL. Uses hyper's default otherwise` I also wasn't able to find any information based on this comment that was in the code.

**Comment by sezan92** [comment]
I also have not found any info . should we tag someone ?

**Reply by luca992** [reply]
The defaults are here in the docs
https://docs.rs/hyper/latest/hyper/client/conn/http2/struct.Builder.html
